### PR TITLE
Small edge case fixes

### PR DESCRIPTION
**Pull request by thyo9470**
Small bug fixes

### CODE REVIEW

**Comment by thyo9470** [comment]
Wait for the other 2 merge requests before merging this one

**Comment by JoyMace** [comment]
These fixes may have been included in my most recent pull request that was already merged.

File: code/routers/user/trip.js [hunk 1]
```diff
@@ -198,7 +198,7 @@ router.get('/user-trips', function(req, res) {
   
   // checks if user is logged in or external request
   if ('userID' in req.query){
-    query['userID'] = req.query.userID;
```
Comment (JoyMace) [inline code/routers/user/trip.js:201]: Should lines 200 and 203 match?

File: code/routers/user/profile.js [hunk 2]
```diff
@@ -65,7 +65,7 @@ router.get('/profile', (req, res) => {
               });
             }
             else{
-              userData['donationCount'] = user[0].count;
+              userData['donationCount'] = user[0] ? user[0].count : 0;
```
Comment (JoyMace) [inline code/routers/user/profile.js:68]: This is already on master

File: code/routers/user/profile.js [hunk 3]
```diff
@@ -55,7 +55,7 @@ router.get('/profile', (req, res) => {
           });
         }
         else{
-          userData['tripsCount'] = user[0].count;
+          userData['tripsCount'] = user[0] ? user[0].count : 0;
```
Comment (JoyMace) [inline code/routers/user/profile.js:58]: This is already on master

File: code/routers/user/profile.js (hunk 2 above)
Comment (thyo9470) [reply code/routers/user/profile.js:68]: It should be fine then. It shouldn't cause any problems, but good to know! thanks!

File: code/routers/user/profile.js (hunk 3 above)
Comment (thyo9470) [reply code/routers/user/profile.js:58]: same as what I said above, but thanks for the hands up :)

**Pull request merged**